### PR TITLE
Correct return type of methods and typo correction.

### PR DESCRIPTION
**Pull request by saurabh-parekh**
### Description

Correct return type of methods and change return type.

### Fixed Issues (if relevant)

N/A

### Manual testing scenarios

N/A

### Contribution checklist
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

**Comment by saurabh-parekh** [comment]
@chirag-wagento I will create forwardport and backport request by own after merge with magento code. Please don't go in hurry for creating other developer's forwardport and backport pull request. We all are aware so we can create our own pull request at right time.Be professional.

File: app/code/Magento/CatalogRule/Controller/Adminhtml/Promo/Widget/CategoriesJson.php [hunk 1]
```diff
@@ -32,7 +33,7 @@ public function __construct(Context $context, Registry $coreRegistry)
     /**
      * Initialize category object in registry
      *
-     * @return Category
+     * @return Category|Boolean
```
Comment (orlangur) [inline app/code/Magento/CatalogRule/Controller/Adminhtml/Promo/Widget/CategoriesJson.php]: How this method can return `Boolean` class?

**Review by orlangur — CHANGES_REQUESTED** [review]

File: app/code/Magento/CatalogRule/Controller/Adminhtml/Promo/Widget/CategoriesJson.php (hunk 1 above)
Comment (saurabh-parekh) [reply app/code/Magento/CatalogRule/Controller/Adminhtml/Promo/Widget/CategoriesJson.php]: @orlangur
Method returns false value at if $rootid not found in category path ids.

`if (!in_array($rootId, $category->getPathIds())) {`
 `       $this->_redirect('catalog/*/', ['_current' => true, 'id' => null]);`
`        return false;`
`   }`

Comment (orlangur) [reply app/code/Magento/CatalogRule/Controller/Adminhtml/Promo/Widget/CategoriesJson.php]: `false` is `bool` primitive type, it has nothing to do with `Boolean` class.

Comment (saurabh-parekh) [reply app/code/Magento/CatalogRule/Controller/Adminhtml/Promo/Widget/CategoriesJson.php]: Okay.Thanks for helping. Should i add simple `boolean` or leave it as it is and remove file from PR?

Comment (orlangur) [reply app/code/Magento/CatalogRule/Controller/Adminhtml/Promo/Widget/CategoriesJson.php]: Add `|bool` and remove unneeded class import.

Comment (saurabh-parekh) [reply app/code/Magento/CatalogRule/Controller/Adminhtml/Promo/Widget/CategoriesJson.php]: I update the file. Please review it.

**Review by orlangur — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @orlangur, thank you for the review.
[ENGCOM-2011](https://jira.corp.magento.com/browse/ENGCOM-2011) has been created to process this Pull Request

**Comment by magento-engcom-team** [comment]
Hi @saurabh-aureate. Thank you for your contribution. 
Changes from your Pull Request will be available with the upcoming 2.2.6 release.

Please, consider to port this solution to 2.3 release line.
You may use [Porting tool](https://github.com/magento/magento2/wiki/Simple-way-to-port-your-solutions-automatically) to port commits automatically.

**Pull request merged**